core/trino-spi/src/main/java/io/trino/spi/cache/ConnectorCacheMetadata.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.cache;
+
+import io.trino.spi.connector.ColumnHandle;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorTableHandle;
+
+import java.util.Optional;
+
+public interface ConnectorCacheMetadata
+{
+    /**
+     * Returns a table identifier for the purpose of caching with {@link CacheManager}.
+     * {@link CacheTableId} together with {@link CacheSplitId} and {@link CacheColumnId}s represents
+     * rows produced by {@link ConnectorPageSource} for a given split. Local table properties
+     * (e.g. rows order) must be part of {@link CacheTableId} if they are present. List of selected
+     * columns should not be part of {@link CacheTableId}. {@link CacheTableId} should not contain
+     * elements that can be derived from {@link CacheSplitId} such as predicate on partition column
+     * which can filter splits entirely.
+     */
```

Comment:
This is too complicated. Why are all those conditions required?

What does it mean for "List of selected columns should not be part of ...", especially in the case of a ConnectorTableHandle representing a pushed-down subplan?

Reply:
> This is too complicated. Why are all those conditions required?

`Local table properties (e.g. rows order) must be part of {@link CacheTableId} ` - Otherwise, we'll get a correctness error

`{@link CacheTableId} should not contain elements that can be derived from {@link CacheSplitId} such as predicate on partition column  which can filter splits entirely` - This is to maximize cache hit rate

> What does it mean for "List of selected columns should not be part of ...", especially in the case of a ConnectorTableHandle representing a pushed-down subplan?

If 2 `ConnectorTableHandle`s are "same" but contain different sets of selected columns, we can create a common subquery with all those columns, so we want them to have the same `CacheTableId`.